corehq/motech/openmrs/tests/test_repeaters.py

```diff
@@ -666,18 +664,17 @@ class VoidedPatientTests(TestCase, TestFileMixin):
     root = os.path.dirname(__file__)
 
     def setUp(self):
-        self.case = CommCareCase.wrap({
-            "domain": DOMAIN,
-            "case_id": "123456",
-            "type": "person",
-            "name": "Eric Idle",
-            "external_id": "94d60c79-59b5-4a2c-90a5-325d6e32b3db",
-        })
+        self.case = CommCareCaseSQL(
+            domain=DOMAIN,
+            case_id="123456",
+            type="person",
+            name="Eric Idle",
+            external_id="94d60c79-59b5-4a2c-90a5-325d6e32b3db",
+            modified_on=datetime.utcnow(),
+            server_modified_on=datetime.utcnow(),
+        )
         self.case.save()
 
-    def tearDown(self):
-        self.case.delete()
-
```

Comment:
this would still be needed right? or else a `self.cleanUp(self.case.delete)` after the `save()` call above?
Or we do that for only couch models? I thought it was something we did for all models because the transaction clean up can fail as well.

Reply:
I think transaction cleanup can only fail in an environment that is using a proxy DB (depending on how the test databases are configured). 

For non-sharded tests it fine (and recommended) to rely on the transaction cleanup rather than manual cleanup for SQL data.

Reply:
No, SQL cases should be deleted on transaction rollback, which is done automatically by Django `TestCase.tearDown`. However, now that you've pointed this out I'm wondering whether Django creates transactions in the shard DBs. I'll check on that before merging.

Reply:
Checked a couple of different ways. Transaction cleanup is working as expected here.

Reply:
thank you both.